internal/services/sentinel/sentinel_watchlist_item_resource.go

```diff
@@ -0,0 +1,177 @@
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IsUUID,
+		},
+		"watchlist_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.WatchlistID,
+		},
+		"fields": {
+			Type:     pluginsdk.TypeMap,
+			Required: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+	}
+}
+
+func (r WatchlistItemResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r WatchlistItemResource) ResourceType() string {
+	return "azurerm_sentinel_watchlist_item"
+}
+
+func (r WatchlistItemResource) ModelObject() interface{} {
+	return &WatchlistItemModel{}
+}
+
+func (r WatchlistItemResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.WatchlistItemID
+}
+
+func (r WatchlistItemResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func:    r.createUpdate,
+	}
+}
+
+func (r WatchlistItemResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Sentinel.WatchlistItemsClient
+			id, err := parse.WatchlistItemID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, id.ResourceGroup, OperationalInsightsResourceProvider, id.WorkspaceName, id.WatchlistName, id.Name)
+			if err != nil {
+				if utils.ResponseWasNotFound(resp.Response) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			watchlistId := parse.NewWatchlistID(id.SubscriptionId, id.ResourceGroup, id.WorkspaceName, id.WatchlistName)
+
+			var fields map[string]interface{}
+			if props := resp.WatchlistItemProperties; props != nil {
+				if itemsKV := props.ItemsKeyValue; itemsKV != nil {
+					fields = itemsKV.(map[string]interface{})
+				}
+			}
+			model := WatchlistItemModel{
+				WatchlistID: watchlistId.ID(),
+				Name:        id.Name,
+				Fields:      fields,
+			}
+
+			return metadata.Encode(&model)
+		},
+	}
+}
+
+func (r WatchlistItemResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Sentinel.WatchlistItemsClient
+
+			id, err := parse.WatchlistItemID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if _, err := client.Delete(ctx, id.ResourceGroup, OperationalInsightsResourceProvider, id.WorkspaceName, id.WatchlistName, id.Name); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r WatchlistItemResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func:    r.createUpdate,
```

Comment:
(same here) this should be a separate method, which in the case of Update retrieves and conditionally patches the resource